### PR TITLE
Bug 1212403 - [TV 2.5] Apply visual spec to the client-side page of T…

### PR DESCRIPTION
**Pull request by luke-chang**
…V remote control

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=2b75e4980b0a5221d335cb58f5cc888eadab0f2d

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=983ffbd67db63213e963c0abe8e14c6524f23883

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=8333a4cc16d4b9830b631247279db32c5db0d21e

File: tv_apps/remote-control-client/js/pairing.js [hunk 1]
```diff
@@ -10,54 +10,181 @@
...
+
+      // allowed function keys: backspace, shift, del and F5
+      if ([8, 16, 46, 116].indexOf(c) < 0) {
+        evt.preventDefault();
+        return;
+      }
+    }
+  };
+
+  function init() {
```
Comment (rexboy7) [inline tv_apps/remote-control-client/js/pairing.js]: Can you move `init()` to the the first function?

File: tv_apps/remote-control-client/js/pairing.js [hunk 2]
```diff
@@ -10,54 +10,181 @@
...
+    var btnSubmit = document.getElementById('connect');
+    var btnRestartPairing = document.getElementById('restart-pairing');
+    var pinCodeInput = new PinCodeInput(
+      document.getElementById('input-mask'),
+      [].slice.call(document.querySelectorAll('.code')),
```
Comment (rexboy7) [inline tv_apps/remote-control-client/js/pairing.js]: We can consider `Array.from()` but it's up to you.

File: tv_apps/remote-control-client/js/pairing.js [hunk 3]
```diff
@@ -10,54 +10,181 @@
...
+      return function(evt) {
+        self.showMask(index);
+        evt.preventDefault();
+      };
+    };
 
-    codes.forEach(function(code, index) {
-      code.value = '';
-      code.placeholder = '-';
+    for(var i = 0; i < codes.length; i++) {
```
Comment (rexboy7) [inline tv_apps/remote-control-client/js/pairing.js]: `foreach()`?

File: tv_apps/remote-control-client/js/pairing.js (hunk 2 above)
Comment (luke-chang) [inline tv_apps/remote-control-client/js/pairing.js]: It's ES6 that we can't use in the client-side page. T_T

File: tv_apps/remote-control-client/js/pairing.js [hunk 4]
```diff
@@ -10,54 +10,181 @@
...
+
+      if (evt.ctrlKey || evt.altKey || evt.metaKey) {
+        return;
+      }
+
+      // backspace to previous field
+      if (c == 8 && this._mask.value === '') {
+        if (this._index > 0) {
+          this.writeBack();
+          this.setCode(this._index - 1, '');
```
Comment (rexboy7) [inline tv_apps/remote-control-client/js/pairing.js]: Please add comment to describe why erasing the previous field.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=b26bf44af938308fcac0d938021a9a764fcde4a6

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=77fb40adc20c6cb0d24dcf199df5aa935e6544ce

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=c010e247fe3784656ba40011661e2898a98b99e8

**Pull request merged**